frontend/src/components/CreatePostModal.jsx

```diff
@@ -186,21 +285,25 @@ class CreatePostModal extends Component {
 			$imagePreview = (<img className="imgPreview" src={imagePreviewUrl} alt="A preview of what you uploaded"/>);
 		}
 		
+		let $modalHeader = (<h3>Create Post</h3>);
+		if (this.props.isEdit) {
+			$modalHeader = (<h3>Edit Post</h3>);
+		}
 	
 		return(
  				<Modal 
- 					trigger={<Button fluid icon onClick={() => this.setState({showModal: true})}> <Icon name="send"/> Create Post </Button>}
-					open={this.state.showModal}
+ 					trigger={this.props.modalTrigger}
+					open={this.props.showModal}
 					onClose={this.closeModal}
  					className={"createPostModal"}
  				>
-					<Modal.Header className='createPostHeader'> <h3> Create Post </h3> </Modal.Header>
+					<Modal.Header className='createPostHeader'> {$modalHeader} </Modal.Header>
 					<Modal.Content className="postModalContent">
 					
 					{this.state.createPostPageOne ?
 					<span>
 					<span className="profileBubbleInModal">
-						<ProfileBubble 	username={this.props.storeItems.username} 
+						<ProfileBubble 	username={this.props.storeItems.username || Cookies.get("username")} 
```

Comment:
Yes but reducers no have ;(

Reply:
what information do you need after logging in, the login endpoint currently returns the full author id: http://host/author/XXXX, it can return more info in it is needed.